src/context/AppProvider/AppContext.tsx

```diff
@@ -105,58 +103,53 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useEffect(() => {
     if (!wallet) return
     ;(async () => {
-      const chainIds = Array.from(supportedChains).filter(chainId =>
+      let chainIds = Array.from(supportedChains).filter(chainId =>
         walletSupportsChain({ chainId, wallet, isSnapInstalled }),
       )
+
+      const accountMetadataByAccountId: AccountMetadataById = {}
       const isMultiAccountWallet = wallet.supportsBip44Accounts()
       for (let accountNumber = 0; chainIds.length > 0; accountNumber++) {
         // only some wallets support multi account
         if (accountNumber > 0 && !isMultiAccountWallet) break
+
         const input = { accountNumber, chainIds, wallet }
-        const accountMetadataByAccountId = await deriveAccountIdsAndMetadata(input)
-        const accountIds: AccountId[] = Object.keys(accountMetadataByAccountId)
+        const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
+        const accountIds = Object.keys(accountIdsAndMetadata)
+
+        Object.assign(accountMetadataByAccountId, accountIdsAndMetadata)
+
         const { getAccount } = portfolioApi.endpoints
-        const opts = { forceRefetch: true }
-        // do *not* upsertOnFetch here - we need to check if the fetched account is empty
         const accountPromises = accountIds.map(accountId =>
-          dispatch(getAccount.initiate({ accountId }, opts)),
+          dispatch(getAccount.initiate({ accountId }, { forceRefetch: true })),
```

Comment:
👀

Reply:
we good here? just moved the opts inline instead of separate variable